src/main/java/server/HTTPServer.java

```diff
@@ -27,11 +32,21 @@ private String getPath(ArrayList<String> request) {
         return request.get(0).split("\\s+")[1];
     }
 
-    private void writeResponse(String path, ReadableWriteable readerWriter) throws IOException {
-        if (paths.contains(path)) {
-            readerWriter.writeLine("HTTP/1.1 200 OK\r\nContent-Length:0\r\n");
-        } else {
-            readerWriter.writeLine("HTTP/1.1 404 Not Found\r\nContent-Length:0\r\n");
+    private Response route(String path) {
+        switch (path) {
+            case "/simple_get_with_body" -> {
+                return new Response(this.protocol, "200 OK", "Content-Length:11", "Hello world");
+            }
+            case "/simple_get", "/" -> {
+                return new Response(this.protocol, "200 OK", "Content-Length:0", "");
+            }
+            default -> {
+                return new Response(this.protocol, "404 Not Found", "Content-Length:0", "");
+            }
         }
     }
+
+    private String format(Response response) {
```

Comment:
**question:** should this class know about the low-level details of assembling the response?

Reply:
I don't think that it should, but I'm a bit stuck on where to put it. 

Things I've considered:
- Creating a `Formatter` class (with a single method) to keep that responsibility separate from the `Response` itself. 
-  Adding the `format` method to `ReaderWriter`, but I think it may be better to keep those responsibilities separate as well. 

It seemed strange to inject a `Formatter` into the `HTTPServer` on instantiation, or when the `call` method is invoked because currently, the only thing that's being injected is the `ReaderWriter`. That seems a bit asymmetrical because the `ReaderWriter` is a higher-level class than a `Formatter`.

Reply:
Sounds like you've already recognized it as a smell, which is a good start! I also agree that those options don't seem to fit the bill but it's good that you're thinking in terms of responsibilities.

This looks like an example of "Feature Envy" which often pops up in scenarios just like this: moving some data fields off into their own class, leaving its old class to start asking for those fields. Maybe searching around for how you'd normally address that can help guide you to another option?

Reply:
(If you end up feeling like you're still completely stuck, we can definitely hop on a call and chat more)

Reply:
Thanks! It seems like the usual solution for Feature Envy is to move the method accessing the class's data into the class itself in order to keep things encapsulated. 

I'm curious whether moving `format` will give the Response class too much responsibility. Is there a way to balance encapsulating the data while also separating responsibilities?